addons/website_sale/static/src/interactions/product/product_variant_preview.js

```diff
@@ -75,30 +67,97 @@ export class ProductVariantPreview extends Interaction {
      *
      * @returns {void}
      */
-    _updateVariantPreview() {
-        this._resetDisplay();
-        const containerWidth = this.el.offsetWidth;
+    _updateVariantPreview(attributePreviewer, attributePreviewerValues) {
+        const { containerWidth, ptavs, ptavCount, offsetWidthPTAVS } = attributePreviewerValues;
+        this._resetDisplay(attributePreviewer);
         let usedWidth = 0;
-        this.displayedPTAVCount = 0;
-        for (const ptav of this.ptavs) {
-            // Remove d-none to be able to get width.
-            ptav.classList.remove('d-none');
-            usedWidth += ptav.offsetWidth + this.margin;
-            this.displayedPTAVCount++;
+        let displayedPTAVCount = 0;
+        for (const ptav of ptavs) {
+            ptav.classList.remove("d-none");
+            usedWidth += offsetWidthPTAVS.get(ptav) + this.margin;
+            displayedPTAVCount++;
             const remainingSpace = containerWidth - usedWidth;
-            const isLastPTAV = ptav === this.ptavs[this.ptavs.length - 1];
-            const hasHiddenPtavs = isLastPTAV && this.ptavCount > this.displayedPTAVCount;
+            const isLastPTAV = ptav === ptavs[ptavs.length - 1];
+            const hasHiddenPtavs = isLastPTAV && ptavCount > displayedPTAVCount;
             if (usedWidth >= containerWidth || hasHiddenPtavs) {
-                this._showHiddenPTAVsElement(ptav, remainingSpace);
+                this._showHiddenPTAVsElement(
+                    attributePreviewerValues, ptav, remainingSpace, displayedPTAVCount,
+                );
                 break;
             }
         }
     }
+
+    /**
+     * Schedules and batches updates for all active `ProductVariantPreview` elements
+     * to refresh their variant previews efficiently.
+     *
+     * Uses `requestAnimationFrame` to ensure that updates occur in sync with the browser’s
+     * rendering cycle, preventing redundant or frequent recalculations (trigger by offsetWidth).
+     */
```

Comment:
Also take the opportunity to explain why we don't target the element itself in the interaction